storage/replica_command.go

```diff
@@ -605,45 +639,30 @@ func (r *Replica) RangeLookup(batch engine.Engine, args proto.RangeLookupRequest
 			if err != nil {
 				return reply, nil, err
 			}
-			kvs = []proto.KeyValue{{Key: key, Value: *val}}
-		}
-	}
 
-	if len(kvs) == 0 {
-		// No matching results were returned from the scan. This could
-		// indicate a very bad system error, but for now we will just
-		// treat it as a retryable Key Mismatch error.
-		err := proto.NewRangeKeyMismatchError(args.Key, args.EndKey, r.Desc())
-		log.Errorf("RangeLookup dispatched to correct range, but no matching RangeDescriptor was found. %s", err)
-		return reply, nil, err
+			// If the intent is not a deletion, return its value.
+			if val != nil {
+				kvs = []proto.KeyValue{{Key: key, Value: *val}}
+				rds = nil
+			}
+		}
 	}
 
-	// Decode all scanned range descriptors, stopping if a range is encountered
-	// which does not have the same metadata prefix as the queried key.
-	rds := make([]proto.RangeDescriptor, len(kvs))
-	for i := range kvs {
+	// Decode all scanned range descriptors which haven't been unmarshaled yet.
+	for i, l := len(rds), len(kvs); i < l; i++ {
```

Comment:
because the descriptor will still be appended even if unmarshalling into it failed. extreme edge case which is harmless if error checking is done properly, but still.